rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -1713,6 +1720,35 @@ public JobIdData reSubmit(String sessionId, String jobId, PathSegment pathSegmen
         return mapper.map(newJobId, JobIdData.class);
     }
 
+    @Override
+    public WorkflowDescription getDescription(String sessionId, String jobId) throws IOException, RestException {
+        Scheduler scheduler = checkAccess(sessionId, "/scheduler/jobs/" + jobId + "/variables");
+        SchedulerSpaceInterface space = getSpaceInterface(sessionId);
+
+        try {
+            String jobXml = scheduler.getJobContent(JobIdImpl.makeJobId(jobId));
+            Job job;
+            try (InputStream tmpWorkflowStream = IOUtils.toInputStream(jobXml, Charset.forName(FILE_ENCODING))) {
+                job = JobFactory.getFactory().createJob(tmpWorkflowStream, null, null, scheduler, space);
+            }
+            WorkflowDescription workflowDescription = new WorkflowDescription();
+            workflowDescription.setName(job.getName());
+            workflowDescription.setProjectName(job.getProjectName());
+            workflowDescription.setDescription(job.getDescription());
+            workflowDescription.setVariables(job.getVariables());
+            Map<String, String> genericInfo = job.getGenericInformation();
+            workflowDescription.setGenericInformation(genericInfo);
+            if (genericInfo != null) {
+                workflowDescription.setBucketName(genericInfo.get(BUCKET_NAME_GI));
+                workflowDescription.setIcon(genericInfo.get(WORKFLOW_ICON_GI));
+                workflowDescription.setDocumentation(genericInfo.get(DOCUMENTATION_GI));
```

Comment:
Same here what if the job has no GI called 'bucket name' ? same for workflow icon and documentation

Reply:
in that case, it is null, what is the problem ?

Reply:
example:

`curl -H "sessionid:$sessionid" -X GET http://mage.activeeon.com:8080/rest/scheduler/jobs/66/description
{"name":"Variables_Propagation","projectName":"1. Basic Workflows","description":"\n     Demonstrate how variables can be read/written to in different script tasks. \n  ","variables":{"numberOfLights":{"name":"numberOfLights","value":"2","model":"PA:LIST(1,2,3,4,5)"},"minRequiredAngle":{"name":"minRequiredAngle","value":"120","model":"PA:INTEGER"},"coveredAnglePerLight":{"name":"coveredAnglePerLight","value":"90","model":"PA:INTEGER"},"currentCoverage":{"name":"currentCoverage","value":"0","model":"PA:INTEGER"}},"genericInformation":{"bucketName":"basic-examples","workflow.icon":"/automation-dashboard/styles/patterns/img/wf-icons/variables_propagation.png","group":"public-objects"},"bucketName":"basic-examples","icon":"/automation-dashboard/styles/patterns/img/wf-icons/variables_propagation.png","documentation":null}`